dev-tools/omdb/src/bin/omdb/nexus.rs

```diff
@@ -1556,6 +1556,62 @@ fn print_task_details(bgtask: &BackgroundTask, details: &serde_json::Value) {
                 }
             }
         }
+    } else if name == "blueprint_loader" {
+        #[derive(Deserialize)]
+        struct BlueprintLoaderStatus {
+            target_id: Uuid,
+            time_created: DateTime<Utc>,
+            status: String,
+            enabled: bool,
+        }
+
+        match serde_json::from_value::<BlueprintLoaderStatus>(details.clone()) {
+            Err(error) => eprintln!(
+                "warning: failed to interpret task details: {:?}: {:?}",
+                error, details
+            ),
```

Comment:
Wondering if there's a way we can incorporate a check that this deserializes correctly. I guess to do that we'd need a test which kicks off a blueprint execution and ensures that it deserializes correctly.

Reply:
We could definitely do more to reify the interface between tasks' status messages and omdb.  In an ideal world they'd show up in the OpenAPI spec and you wouldn't have to do this at all.  When we added background tasks the obvious way to do that was with an enum with variants for each task, but that seemed unwieldy on a bunch of levels.  But yeah, the more we add here, the more annoying this is.

Reply:
Oh yeah I think the common layer should be serde_json::Value. But the last and first things we do can be this specific type.

I had to do something similar for the update engine—in there, I implemented a scheme where each event report carries the name of the spec it is associated with. That worked reasonably well.

Reply:
(If this is too hard feel free to defer this, I'm going to be making some changes in this area soon)

Reply:
Now that you mention the idea of a separate name specifying the schema, I wonder if we could first-class that in the JsonSchema (i.e., "if this string property is X, then this other property is of type Y") so that on the other end Progenitor could deserialize to the specific type.

Reply:
Yeah, and maybe the answer is to have the JsonSchema for this type expose it the same way it would a tagged enum?